### DISCUSSION TITLE
Affliction caused by being in water

### DISCUSSION THREAD

**Discussion by ViktorShield** [FakeFishGames/Barotrauma#7758] [Code questions and modding support]
I'm planning on making a hypothermia update to Advanced Medicine mod. How do I make it so the affliction is gained while touching the water or being submerged in it?

**Comment by StruggleDuck** [comment]
maybe start applying it if the character is swimming?

**Reply by ViktorShield** [reply]
Then the character will only gain the affliction if the water level is high. Also, if he's not in the swimming mode for some reason, he'll become invulnerable to it. Even more, this cannot be per-limb

**Reply by StruggleDuck** [reply]
I just thought of something, when wadding through water your character moves slower. Maybe you can use that to see if someone is in water

**Reply by StruggleDuck** [reply]
Furthermore, depending on how slowed they are you can figure out how deep the water is

**Reply by ViktorShield** [reply]
I have no idea how to read that info
